### PR TITLE
Feature/haptic projection

### PR DESCRIPTION
**Pull request by jthrun**
Work to do:

- [x] Integrate with changes from Enhanced Video Streaming APIs PR #471 
- [ ] Sync functionality with iOS haptic interface manager
- [x] Unit tests

### CODE REVIEW

**Comment by jthrun** [comment]
Unit tests are done.
Can't integrate with VirtualDisplayEncoder until that PR is merged.
Not sure that hit testing is needed in the Android library as is in the iOS library.

@joeygrover, ready for code review, please add others as I'm not sure who else should review.

File: sdl_android/src/main/java/com/smartdevicelink/haptic/HapticInterfaceManager.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+     *
+     * @param hapticData
+     *          Rect data indicating "focusable" screen elements or areas
+     */
+    public void setHapticData(List<HapticRect> hapticData) {
+        userHapticData = hapticData;
+        SdlProxyBase proxy = proxyHolder.get();
+        if (proxy != null) {
+            SendHapticData msg = new SendHapticData();
+            msg.setCorrelationID(CorrelationIdGenerator.generateId());
```
Comment (brettywhite) [inline sdl_android/src/main/java/com/smartdevicelink/haptic/HapticInterfaceManager.java]: if you don't need to keep track of this ID, this line is no longer needed

File: sdl_android/src/main/java/com/smartdevicelink/haptic/HapticInterfaceManager.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+     *          the root or parent View
+     */
+    public void refreshHapticData(View root) {
+        SdlProxyBase proxy = proxyHolder.get();
+        if ((userHapticData == null) && (proxy != null)) {
+            List<HapticRect> hapticRects = new ArrayList<>();
+            findHapticRects(root, hapticRects);
+
+            SendHapticData msg = new SendHapticData();
+            msg.setCorrelationID(CorrelationIdGenerator.generateId());
```
Comment (brettywhite) [inline sdl_android/src/main/java/com/smartdevicelink/haptic/HapticInterfaceManager.java]: if you don't need to keep track of this ID, this line is no longer needed

File: sdl_android/src/main/java/com/smartdevicelink/haptic/HapticInterfaceManager.java (hunk 1 above)
Comment (jthrun) [reply sdl_android/src/main/java/com/smartdevicelink/haptic/HapticInterfaceManager.java]: OK, I will remove them.

**Comment by jthrun** [comment]
Integrated with the recently merged enhanced video streaming changes.  I believe this is ready to merge.

**Comment by brettywhite** [comment]
@jthrun is there a way to have the unit test align with how the other tests currently are?

**Comment by jthrun** [comment]
@brettywhite it is possible, it'll be harder to set up and verify without mockito since it will require using actual View instances and writing a test ISdl interface instance.

**Comment by brettywhite** [comment]
@jthrun Its all good, we can leave it there.

**Comment by jthrun** [comment]
@brettywhite OK, I did try when I first wrote the unit test to get mockito to work within the androidTest tree but couldn't make it happy.  Seems like mockito isn't supported there, at least without a lot of unofficial hacking.

**Pull request merged**